gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/internal/ByUserOriginSelector.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.services.internal;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.kork.web.selector.SelectableService;
+import com.netflix.spinnaker.kork.web.selector.ServiceSelector;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ByUserOriginSelector implements ServiceSelector {
```

Comment:
this class maybe should live in kork with the other selectors, making it live closer for now while we test this out